src/qlippermodel.cpp

```diff
@@ -243,3 +246,22 @@ void QlipperModel::timer_timeout()
     m_timer->start();
 #endif
 }
+
+bool QlipperModel::confirm_clear()
```

Comment:
1. please, do not put user interaction into the model (we should use Qt the model/view architecture). IMO the message box logic should go into `QlipperSystray` (connect action to lambda or `QlipperSystray` method which only calls the `clearHistory()` in case it is confirmed)
2. consider using the `QMessageBox::question()`
3. don't use untranslatable static strings for user interaction (mostly everything shown in UI should be translatable using `QObject::tr()`